c10/util/BFloat16.h

```diff
@@ -5,21 +5,37 @@
 
 #include <c10/macros/Macros.h>
 #include <cmath>
+#include <cstring>
 
 namespace c10 {
 
 namespace detail {
+  inline bool isSmallEndian() {
+    int num = 1;
```

Comment:
you should test this statically.

Reply:
You link https://stackoverflow.com/questions/1583791/constexpr-and-endianness presumably because you wanted to implement this using `constexpr`. But I don't think we actually are requiring you to implement it with constexpr. All we are asking is that there isn't actually a conditional check in the generated machine code. I don't know if the current implementation you have here, it is managed to be compiled away; if it is, that would be sufficient for me (it's a good thing to put in a comment.)